miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/UrlToFileInfoParser.kt

```diff
@@ -0,0 +1,40 @@
+package com.rakuten.tech.mobile.miniapp.storage
+
+import java.io.File
+
+private const val KEY_MAP_PUB = "map-published"
+
+internal class UrlToFileInfoParser {
+
+    /**
+     * Returns the path between versionId to file name e.g. for a given source,
+     * "https://host.os.net/map-published/min-872f9172-804f-44e2-addd-ed612170dac9/
+     * >>ver-6181004c-a6aa-4eda-b145-a5ff73fc4ad0/foo/bar/asset-manifest.json",
+     * "/foo/bar/" will be returned.
+     * Returns empty String when not found.
+     */
+    @Suppress("MagicNumber")
+    fun getFilePath(fileUrl: String): String = fileUrl.split(File.separator).run {
+        val versionIndex = indexOf(KEY_MAP_PUB)
+        return when {
+            // If versionIndex = -1, map-published was not found, then this source is invalid.
+            versionIndex < 0 -> ""
+            else -> {
+                var path = "/"
+                // The third string after "version" is the beginning of path.
+                subList(versionIndex + 3, lastIndex).forEach { path += "$it/" }
```

Comment:
You can get rid of magic number by define property for `min` then instead of 3, we could get the length of it.

Reply:
I ended up removing the warning by extracting it out as a file-private constant value. My initial attempt was to keep the intelligence within the method. Looks like there isn't a nicer workaround. Thanks..!